Frontend/src/Components/Dashboard/Dashboard.js

```diff
@@ -1,44 +1,87 @@
-import { Grid, GridItem, HStack, VStack, Select, Box } from "@chakra-ui/react";
-import React, { useState, useLayoutEffect } from "react";
+import { Box, Grid, GridItem, HStack, Select } from "@chakra-ui/react";
+import { useEffect, useReducer, useState } from "react";
+import BatteryCells from "../BatteryCells/BatteryCells";
 import FaultsView from "../Faults/FaultsView";
 import DataView from "../GeneralData/DataView";
-import BatteryCells from "../BatteryCells/BatteryCells";
-import PPC_MPPT from "../PPC_MPPT/PPC_MPPT";
 import MiniMap from "../MiniMap/MiniMap";
-import BatteryGraph from "../Graph/BatteryGraph";
-import PowerGraph from "../Graph/PowerGraph";
-import TemperatureGraph from "../Graph/TemperatureGraph";
-import CustomGraph from "../Graph/CustomGraph";
+import GraphContainer from "./GraphContainer";
+
+// prevent accidental reloading/closing
+window.onbeforeunload = () => true;
+
+function reducer([currentQueue, lastTime], newData) {

```

Comment:
A small header for this reducer would be nice

Reply:
Thanks for the reminder; added